Core/Models/PointMassSecondOrder.hpp

```diff
@@ -74,6 +74,14 @@ namespace gtfo{
             / Base:: parameters_.mass) * (Base::position_ - VectorN::Constant(Base::parameters_.virtual_spring_zero_position)) + force_input / Base::parameters_.mass;
         }
 
+        void UpdateParameters(const Scalar& mass, const Scalar& damping, const Scalar& stiffness, const Scalar& virtual_spring_zero_position) {
+            Base::parameters_.mass = mass;
+            Base::parameters_.damping = damping;
+            Base::parameters_.stiffness = stiffness;
+            Base::parameters_.virtual_spring_zero_position = virtual_spring_zero_position;
+        }
+
+
```

Comment:
For abstractness, I think this function could just accept a parameters struct, rather than the items within the parameter. If the definition of the parameters struct changes, then this method would not correctly update all the values, creating a hard-to-find bug. 

Since parameters are relevant at the `PointMassBase` level, I would suggest putting this method there instead. Furthermore, the update parameters function should call `SetStateTransitionMatrices`.